src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs

```diff
@@ -99,6 +137,22 @@ public ResultWrapper<GethLikeTxTrace> debug_traceTransactionByBlockhashAndIndex(
     {
         using CancellationTokenSource timeout = BuildTimeoutCancellationTokenSource();
         CancellationToken cancellationToken = timeout.Token;
+
+        // HasStateForBlocks Start
+        SearchResult<Block> blockSearch = _blockFinder.SearchForBlock(new BlockParameter(blockhash));
+        if (blockSearch.IsError)
+        {
+            return ResultWrapper<GethLikeTxTrace>.Fail(blockSearch);
+        }
+
+        Block block = blockSearch.Object!;
```

Comment:
Add a check in case if the searchResult has errors i.e
```c#
        if (searchResult.IsError || searchResult.Object is null)
            return ResultWrapper<IReadOnlyList<SimulateBlockResult<TTrace>>>.Fail(searchResult);
           
```